### PR TITLE
[UI] Add intersection observer for components and relationship tabs

### LINKED ISSUE #11471: [Server] Registry Enhancements
### Current Behavior
- [x] 1. Update the API endpoints to retrieve only the count of components and relationships for each model, rather than returning all data.
    - [x] Fix the bug where both the component count and relationship count are showing zero,
- [ ] 2. Remove duplicated data that's currently being returned from the endpoint.
- [ ] 3. Add query parameters for the `meshmodel/models`, `meshmodels/components`, `meshmodels/relationships`, and `meshmodels/registrants` endpoints.
    - [ ] Add a query parameter like `duplication-data=true` to fetch information about duplicated entities. This will allow the UI to filter and highlight duplicate entities.
- [ ] 4. Remove the aggregation of models in the registrants tab once the data duplication is resolved from the endpoint.
- [ ] 5. Add ID-based `GET` endpoints for all Registry requests to simplify the selection of entities. #12361
    - Currently, selection is based on a combination of name and version.
    - Possibly related PR: https://github.com/meshery/meshkit/pull/538.
- [ ] 6. [UI] Ensure the `meshmodel` details component respects the current level of navigation in the tree structure. (depends on 1st issue of this list)
    - [ ] At the top level of a model, display the aggregate count of components across all versions of that model.

### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

### To Reproduce
1. Go to [https://playground.meshery.io/settings?settingsCategory=Registry](https://playground.meshery.io/settings?settingsCategory=Registry)

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by amitamrutiya**
**Notes for Reviewers**

This PR Partial fixes #11471

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by Yashsharma1911 — COMMENTED** [review]
Thankyou can you please add a video after changes?

**Review by Yashsharma1911 — APPROVED** [review]
After video we can merge, for now LGTM

**Comment by amitamrutiya** [comment]
[Screencast from 2024-09-01 12-49-43.webm](https://github.com/user-attachments/assets/57ae3a7b-5cf1-4a72-ab2a-91bc6c23f13a)

**Comment by Yashsharma1911** [comment]
> Screencast.from.2024-09-01.12-49-43.webm

In video why relationship count it wrong? it shows 234 relationships on tab and on tree only 25?

**Comment by Ashparshp** [comment]
Thank you @amitamrutiya, let's discuss this PR during the Meshery dev meeting call, if you would. Adding it as an agenda item to the meeting minutes.

Timing: 7:30 PM IST / 9:00 AM CT (Wednesday)
Meeting minutes: [Google Doc](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit)
Meeting link: [meet.layer5.io/meshery](https://meet.layer5.io/meshery)

**Comment by Yashsharma1911** [comment]
@amitamrutiya LGTM now, however tests are failing

**Comment by amitamrutiya** [comment]
@Yashsharma1911 Test is failing because of different reasons. It has failed in all of the recent PRs.

File: ui/api/meshmodel.js [hunk 1]
```diff
@@ -20,7 +19,6 @@ const CATEGORIES_ENDPOINT = '/api/meshmodels/categories';
 
 /** @type {pageOptions} */
 const defaultOptions = {
-  paginated: false,
```
Comment (Yashsharma1911) [inline ui/api/meshmodel.js:23]: Removing paginated from here will affect where ever this defaultOptions var is used, you sure about removing this, why we need to make change here?

Comment (amitamrutiya) [reply ui/api/meshmodel.js:23]: I checked everywhere in the frontend, backend, and API documentation, and we are not using the `paginated` parameter anywhere.

**Review by Yashsharma1911 — APPROVED** [review]
LGTM

**Comment by Yashsharma1911** [comment]
@amitamrutiya let's keep an eye on staging

**Pull request merged**